### ISSUE TITLE
Ellipsis instead of three dots

### ISSUE THREAD

**Issue by Valdnet** [joomla/joomla-cms#38630]
The spelling of the ellipsis should be corrected in all text strings. Currently, three dots `...` are used which is a spelling mistake. The ellipsis character of the form `…` should be used.

**Comment by ReLater** [comment]
I would prefer a language string for the code places where the three dots are used (e.g. in truncate methods, especially truncateComplex). Then everyone can decide for themselves what they want to use. Empty string, icon, 3 dots, ellipsis, whatever.

But I'm against a change in core like hard-coded ellipsis instead of 3 dots.

**Comment by brianteeman** [comment]
The style guide says it should be an ellipsis (https://developer.joomla.org/en-gb-user-interface-text-guidelines/punctuation.html)

Also a screen reader understands an ellipsis but doesnt understand three dots

**Comment by ReLater** [comment]
> Also a screen reader understands an ellipsis but doesnt understand three dots

Set the default of my proposed language string to an ellipsis.

Method truncateComplex sets the 3 dots/ellipsis outside of the last HTML tag; e.g. after last `<p></p>`.

I think a language string would be the best solution for all needs.

**Comment by brianteeman** [comment]
> Method truncateComplex sets the 3 dots/ellipsis outside of the last HTML tag; e.g. after last <p></p>.

It shouldn't do that. Can you post an example please.

**Comment by ReLater** [comment]
That's the behavior since ever.

```php
$text = '<p>Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo Hello hallo </p>';
$introtext_limit = 35;

echo HTMLHelper::_('string.truncateComplex', $text, $introtext_limit);
```

`==>`

`<p>Hello hallo Hello hallo Hello</p>...`

**Comment by brianteeman** [comment]
> That's the behavior since ever.

then its always been a bug

**Comment by N6REJ** [comment]
@brianteeman apparently this was introduced in J! 2.5 https://blog.crosborne.uk/truncstring-joomla-bug

**Comment by brianteeman** [comment]
thanks - will read it in the morning

**Comment by N6REJ** [comment]
looking @ the [code](https://github.com/joomla/joomla-cms/blob/9df5e6c7a7a1c6ec0aed99d6d4a22be048b9b299/libraries/src/HTML/Helpers/StringHelper.php) it looks like it was designed to do that...
```
            // If the new plain text string matches the original plain text string we are done.
            if ($ptString === $htmlStringToPtString) {
                return $htmlString . '...';
            }

            // Get the number of HTML tag characters in the first $maxLength characters
            $diffLength = strlen($ptString) - strlen($htmlStringToPtString);

            if ($diffLength <= 0) {
                return $htmlString . '...';
            }
            ```

**Comment by Valdnet** [comment]
@N6REJ You should also remember about text strings that come with three dots, such as this one:
https://github.com/joomla/joomla-cms/blob/9df5e6c7a7a1c6ec0aed99d6d4a22be048b9b299/language/en-GB/com_media.ini#L101

Additionally, the ellipsis should be separated by a space.

**Comment by ReLater** [comment]
Just to have it mentioned:

You should keep in mind that simple `<p></p>` examples are relatively easy to fix. Find the last closing tag move ... inside.

It becomes more complicated if you work with more complex HTML:

```php
$text = '
<p>Happa heppiu.</p>
<ul>
<li>A1 A1 A1 A1 A1 A1 A1 A1 A1 A1 A1 A1</li>
<li>A2 A2 A2 A2 A2 A2 A2</li>
<li>A3 A4 A5 A6 A7 A8 A9 A10 A11 A12 A13 A14</li>
<li>B1cdsww B1cdsww B1cdsww B1cdsww B1cdsww B1cdsww B1cdsww</li>
<li>B2weqwewq B2weqwewq B2weqwewq B2weqwewq B2weqwewq B2weqwewq</li>
</ul>
';
$introtext_limit = 100;
echo HTMLHelper::_('string.truncateComplex', $text, $introtext_limit);
```

`=>`

```html
<p>Happa heppiu.</p>
<ul>
<li>A1 A1 A1 A1 A1 A1 A1 A1 A1 A1 A1 A1</li>
<li>A2 A2 A2 A2 A2 A2 A2</li>
<li>A3 A4 A5 A6 A7 A8 A9 A10</li></ul>...
```

instead of

```html
<p>Happa heppiu.</p>
<ul>
<li>A1 A1 A1 A1 A1 A1 A1 A1 A1 A1 A1 A1</li>
<li>A2 A2 A2 A2 A2 A2 A2</li>
<li>A3 A4 A5 A6 A7 A8 A9 A10...</li></ul>
```

**Comment by brianteeman** [comment]
It actually happens very rarely in core. But there are solutions available.

**Comment by brianteeman** [comment]
@laoneo it looks like you have addressed this exact bug in dpcalendar

**Comment by laoneo** [comment]
Did I? Not sure what you mean...
